framework/helpers/BaseStringHelper.php

```diff
@@ -104,12 +104,15 @@ public static function dirname($path)
      */
     public static function truncate($string, $length, $suffix = '...', $encoding = null, $asHtml = false)
     {
+        if ($encoding === null) {
+            $encoding = Yii::$app ? Yii::$app->charset : 'UTF-8';
```

Comment:
> We should either introduce a static level for default encoding 

isn't that the job of Yii::$app->charset? It is default to `'UTF-8'` if you configure your application and change it you get a different encoding. Ternary check only makes the default value available if app is not configured.